app/uk/gov/hmrc/gform/LookupLoader.scala

```diff
@@ -102,19 +108,42 @@ class LookupLoader {
     keywords: String,
     priority: String,
     region: String,
+    inEU: String,
+    inEEA: String,
+    inEFTA: String,
     mkLookupType: LocalisedLookupOptions => LookupType
   ): LookupType = {
 
-    type ColumnData = (LookupLabel, LookupLabel, LookupId, LookupKeywords, LookupPriority, LookupRegion)
+    type ColumnData = (
+      LookupLabel,
+      LookupLabel,
+      LookupId,
+      LookupKeywords,
+      LookupPriority,
+      LookupRegion,
+      LookupInEU,
+      LookupInEEA,
+      LookupInEFTA
+    )
 
     val headerDecoder: HeaderDecoder[ColumnData] =
-      HeaderDecoder.decoder(englishLabel, welshLabel, id, keywords, priority, region)(
-        (_: LookupLabel, _: LookupLabel, _: LookupId, _: LookupKeywords, _: LookupPriority, _: LookupRegion)
+      HeaderDecoder.decoder(englishLabel, welshLabel, id, keywords, priority, region, inEU, inEEA, inEFTA)(
```

Comment:
Sorry for late notice, but I don't think this is right.
We should not read `inEU`, `inEEA` and `inEFTA` explicitly. We should just read whole csv row into `Map[String, String]` (where keys are column names) and base a lookup of column value from `CsvCountryCheck ` on this Map. This will give us nice point of failure. Currently if `column` doesn't exists in csv it doesn't crash

Reply:
this method (`readCountries`) produces `CountryLookupInfo` with structure that replicates the csv column and is used for Ajax lookups. It is also used in a lot of places in `match` expressions.  I think if we can't easy make it based on the `Map`.
Shall we create a separate `CountryLookupCheckInfo` for `CsvCountryCheck` only ? It will be based on `Map`.

Reply:
I'would prefer to extend `CountryLookupInfo` to look like this:
```
final case class CountryLookupInfo(
  id: LookupId,
  index: Int,
  keywords: LookupKeywords,
  priority: LookupPriority,
  region: LookupRegion,
  columns: Map[String, String]
) extends LookupInfo
```
I feel that to have one object with entirety data from csv row is a good thing.
I think it should be possible to read into the Map at this point: https://github.com/hmrc/gform-frontend/blob/main/app/uk/gov/hmrc/gform/LookupLoader.scala#L117
(We have `index: Int` which I suppose is row number, so we can read that row)

Reply:
ok, will to it this way, :bow: